saleor/graphql/product/mutations/products.py

```diff
@@ -204,8 +216,19 @@ class Meta:
     @classmethod
     def clean_input(cls, info, instance, data):
         cleaned_input = super().clean_input(info, instance, data)
-        if "slug" not in cleaned_input and "name" in cleaned_input:
-            cleaned_input["slug"] = slugify(cleaned_input["name"])
+        try:
+            cleaned_input = validate_slug_and_generate_if_needed(
+                instance, "name", cleaned_input
+            )
+        except ValidationError:
+            raise ValidationError(
+                {
+                    "slug": ValidationError(
+                        "Slug value cannot be blank.",
```

Comment:
Maby we should use the message from catched exception, in all places when we use `validate_slug_and_generate_if_needed`?